feat(blog): add fcb case components

milljoniaer opened the pull request:
### This adds the new components needed for the fcb case

the stats
<img width="537" alt="image" src="https://user-images.githubusercontent.com/78978542/163345848-d9bc1a96-d240-4098-bd0e-eae250b0e6d5.png">
[preview](https://6257f90a41c03a7096b4fd43--satellytes-website-storybook.netlify.app/?path=/docs/components-content-stats--regular&globals=locale:en;locales.en:English;locales.de:Deutsch)

the full width images
![image](https://user-images.githubusercontent.com/78978542/165262267-ce22ff79-665b-423e-84c6-512283fece61.png)


#### Build Fail
The build failes currently, because there is no ContentfulStats Object published so far. Therefore the query fails.

Maybe we should wait with merging this until we can publish the FCB Case in Contentful.

You can still test this component (also within the fcb case) by running this branch locally

milljoniaer committed:
feat(blog): add stats

milljoniaer committed:
Merge branch 'main' into feat/add-blog-post-stats

ghost commented:
# [Gatsby Cloud](https://www.gatsbyjs.com) Build Report

:triangular_flag_on_post: Your build failed. [See the build logs here](https://gatsbyjs.com/dashboard/f215130d-15e3-4176-9252-e13166fe48de/sites/674be368-adf0-4ac5-8fae-f9a4493adc51/builds/ab7186fd-5a51-47d9-954f-906b604e236f/details)

milljoniaer committed:
chore: margins

georgiee reviewed (approved):
Thanks, looks good. Did you interact with @dschu-lab. I think it's a good idea in terms of contentful integration. I only glimpsed through the files and it looks good 👍

JulianAnthes commented on src/components/legacy/markdown/custom-components.tsx:
```diff
@@ -228,6 +229,20 @@ const customSatellytesComponents = {
+        ))}
+      </Stats>
+    );
+  },
```
I think we can move the rendering part directly into `src/components/pages/blog-post/blog-post.tsx` since the properties of `customComponents` represent html tags that are rendered by the markdown plugin.

milljoniaer replied:
you are right, I fixed it 👍

JulianAnthes commented on src/templates/blog-post.tsx:
```diff
@@ -111,6 +111,15 @@ export const BlogPostPageQuery = graphql`
             }
           }
 
+          ... on ContentfulStats {
+            contentful_id
+            __typename
+            statItems {
+              label
+              value
+            }
+          }
+
```
Welcome `ContentfulStats` 👋

milljoniaer replied:
🥳

JulianAnthes commented on src/components/content/stats/stats.tsx:
```diff
@@ -0,0 +1,47 @@
+  ${up('sm')} {
+    flex-direction: row;
+  }
+`;
```
Lets replace those `div` tags with more meaningful ones, like the [dl tag](https://www.w3schools.com/TAGs/tag_dl.asp):

### Example HTML result
```html
<dl>
  <dt>Team Size</dt>
  <dd>6 people</dd>
  <dt>Start</dt>
  <dd>December 2018</dd>
  <dt>End</dt>
  <dd>Ongoing</dd>
</dl>
```

Since we need to remove the `Wrapper` (`StatsItem` wrapper), we should switch to a css grid for the `StatsWrapper`. 

I have created a small codepen as orientation for you: [Codepen Example](https://codepen.io/dschu/pen/LYegRNv)

milljoniaer replied:
thank you for suggestion, this is convenient, I changed it 👍

JulianAnthes reviewed (changes requested):
Looks like a really great feature! Can't wait to see this released. But we should take the time and improve on the semantic html. 

Moving the stats rendering depends on if it is possible to use it in plain markdown. If it works, we can leave it as is, but if not then it would be better to "lift it one level higher".

milljoniaer committed:
feat: add dl, dt, dd html tags

milljoniaer committed:
chore: move rich text rendering to blog post

milljoniaer commented:
Thank you for your suggestions. I changed them, I also added the storybook deployment label and added the preview link to my description.

Please have another look.

JulianAnthes reviewed (approved):
Thanks for refactoring this into a description list. 
I have nothing more to add, great work! 💪

milljoniaer committed:
Merge branch 'main' into feat/add-blog-post-stats

milljoniaer committed:
chore: pixel

milljoniaer committed:
feat: add FullWidthImages to BlogPosts

milljoniaer commented:
Since the fcb case also requires full width images, I added these also in this branch to have the possibility to see the whole fcb case in action when starting this branch locally.

milljoniaer committed:
chore: remove unused

milljoniaer committed:
Merge branch 'main' into feat/add-blog-post-stats
